pkgs/stdenv/generic/default.nix

```diff
@@ -319,17 +331,15 @@ let
           userHook = config.stdenv.userHook or null;
           __ignoreNulls = true;
 
-          # Inputs built by the cross compiler.
-          buildInputs = buildInputs';
-          propagatedBuildInputs = propagatedBuildInputs';
-          # Inputs built by the usual native compiler.
-          nativeBuildInputs = nativeBuildInputs'
-            ++ lib.optional
-                (hostPlatform.isCygwin
-                  || (crossConfig != null && lib.hasSuffix "mingw32" crossConfig))
-                ../../build-support/setup-hooks/win-dll-link.sh
-            ;
-          propagatedNativeBuildInputs = propagatedNativeBuildInputs';
+          preNativeBuildInputs = lib.elemAt dependencies' 0;
```

Comment:
@0xABAB In some sense the names are more noise than the offsets: the sorts of dependencies are defined by the offsets, and whether they are propogated. We currently use -2 through 1, but in theory the entire space of integers is valid, though after a certain distance from 0 the distinctions collapse.